tools/publishIsolate.ts

```diff
@@ -0,0 +1,57 @@
+import fs from 'fs';
+import path from 'path';
+import childProcess from 'child_process';
+import getNpmPackageVersion from 'get-npm-package-version';
+import { compareVersions } from 'compare-versions';
+
+async function main() {
+  const isolatePackageRoot = './dist/isolate';
+
+  const publishPackageList: string[] = [];
+
+  // get directory listing
+  const files = fs.readdirSync(isolatePackageRoot);
+  for (const file of files) {
+    const filePath = path.join(isolatePackageRoot, file);
+    const stat = fs.statSync(filePath);
+    if (stat.isDirectory()) {
+      // get package.json
+      const packageJsonPath = path.join(filePath, 'package.json');
+      const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+      // get version
+      const version = getNpmPackageVersion(packageJson.name);
+
+      if (version === null) {
+        throw new Error(`Package ${packageJson.name} not found on npm`);
+      }
+
+      // compare version
+      if (compareVersions(version, packageJson.version) < 0) {
+        publishPackageList.push(packageJson.name);
+      }
+    }
+  }
+
+  // publish packages
+  // eslint-disable-next-line compat/compat
+  await Promise.all<void>(publishPackageList.map(packageName => new Promise((resolve, _reject) => {
+    console.log(`Publishing ${packageName}`);
+    const packagePath = path.join(isolatePackageRoot, packageName);
```

Comment:
packagePath will be `dist/isolate/@js-sdsl/xxx`

Reply:
Do you want me to modify the build script so that even the @js-sdsl folder is created?

Reply:
```typescript
const packagePath = path.join(isolatePackageRoot, packageName);
console.log(packagePath);  // output 'dist/isolate/@js-sdsl/xxx' expected 'dist/isolate/xxx'
```